daemon/daemon.go

```diff
@@ -1278,6 +1268,17 @@ func NewDaemon() (*Daemon, *endpointRestoreState, error) {
 	if err := node.ValidatePostInit(); err != nil {
 		log.WithError(err).Fatal("postinit failed")
 	}
+
+	if k8s.IsEnabled() {
```

Comment:
Add a comment indicating why we are doing this here (i.e., the reason behind this PR). This will provide context to future developers.